tests/unit/user_interaction_test.py

```diff
@@ -0,0 +1,76 @@
+from unittest import TestCase
+import user_interaction
+from unittest.mock import patch
+import random
+import string
+
+
+class UserInteractionTest(TestCase):
+    def setUp(self):
+        random.seed(408245472491)
+
+    def get_text_data(self):
+        return ["{", "#", "''", "Cat", "dog", "BIRD", "-1^&&%SS", ""]
+
+    def print_test(self, func, text, test_text=None):
+        with patch("builtins.print") as mocked_print:
+            if test_text != None:
+                func(test_text)
+            else:
+                func()
+            mocked_print.assert_called_with(text)
+
+    def test_show_global_instant(self):
+        test_text_list = self.get_text_data()
+        for test_text in test_text_list:
+            self.print_test(user_interaction.show_global_instant,
+                            f"GLOBAL: {test_text}", test_text)
+
+    def test_show_active_instant(self):
+        test_text_list = self.get_text_data()
+        for test_text in test_text_list:
+            self.print_test(user_interaction.show_active_instant,
+                            f"ACTIVE: {test_text}", test_text)
+
+    def test_show_passive_instant(self):
+        test_text_list = self.get_text_data()
+        for test_text in test_text_list:
+            self.print_test(user_interaction.show_passive_instant,
+                            f"PASSIVE: {test_text}", test_text)
+
+    def test_show_all(self):
+        test_text_list = self.get_text_data()
+
+        user_interaction.save_global(test_text_list[0])
+        user_interaction.save_active(test_text_list[1])
+        user_interaction.save_passive(test_text_list[2])
+        user_interaction.save_passive(test_text_list[3])
+        user_interaction.save_passive(test_text_list[4])
+        user_interaction.save_global(test_text_list[5])
+        user_interaction.save_active(test_text_list[6])
+        user_interaction.save_global(test_text_list[7])
+
+        asserted_text = f"{user_interaction.MessageScope.PASSIVE.name}: {str(test_text_list[2:5])}"
+
+        self.print_test(user_interaction.show_all, asserted_text)
+
+    def test_read_index(self):
+        valid_test_data_list = [random.randint(0, 1000), 0]
+        for i in range(100):
+            for test_data_item in valid_test_data_list:
+                test_to_string = str(test_data_item)
+
+                with patch('builtins.input', return_value=test_to_string):
+                    self.assertEqual(user_interaction.read_index(
+                        test_to_string), test_data_item)
+
+        invalid_test_data_list = [random.randint(-1000, -1), *self.get_text_data(), random.random(), random.choice([
+            True, False]), None]
+        for i in range(100):
```

Comment:
Same issue here (as in previous example).

Also, it's not cool that such values as 'None' and 'True\False' will be tested 100 times. Its enough to test theme once. 
I suggest to move them to a separate list:
invalid_test_data_list_rnd = [random.randint(-1000, -1),  random.random()]
invalid_test_data_list = [*self.get_text_data(), True, False, None]

Reply:
My suggestion is to create 4 cases:
0 - valid case
randint(1,1000) - 100 values - randomly generated valid cases 

[*self.get_text_data(), True, False, None] - invalid cases
random.randint(-1000, -1), random.random() - 100 values - randomly generated invalid cases

Reply:
Now it works. Some upgrades on the next commits.